examples/map/rotate_custom_affine_transform_opencv.py

```diff
@@ -0,0 +1,158 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# # Using a Custom Affine Transform in the Sunpy Map rotate() function
+# #### (with OpenCV)
+# ----------
+# How to construct a custom affine transform for in `map.rotate`.
+# Requires OpenCV (cv2) Python library.
+
+import sunpy.data.sample
+import sunpy.map
+import numbers
+import numpy as np
+import cv2
+
+#############################
+# Rotating a map in sunpy (via `map.rotate()`) uses the `skimage` or `scipy` libraries by default.
+# However, the `method=` argument in `sunpy.map.rotate` can accept a custom function designed
+# to use an external library. Here, we will define an affine transform function
+# with the OpenCV library.
+# Our custom method must have a similar function call to `sunpy.image.transform.affine_transform`
+# and return the same output, namely the rotated, scaled, and transformed image.
+# The required arguments are:
+
+# Parameters
+# ----------
+# image: `numpy.ndarray`
+#     2D image to be rotated
+# rmatrix : `numpy.ndarray` that is 2x2
+#     Linear transformation rotation matrix.
+# order : `int` 0-5
+#     Interpolation order to be used
+# scale : `float`
+#     A scale factor for the image
+# missing : `float`
+#    The value to replace any missing data after the transformation.
+# image_center : tuple, optional
+#     The point in the image to rotate around (axis of rotation).
+#     Defaults to the center of the array.
+# recenter : `bool` or array-like, optional
+#     Move the axis of rotation to the center of the array or recenter coords.
+#     Defaults to `True` i.e., recenter to the center of the array.
+
+
+# NOTE: the required libraries have already been imported globally
+# (see above: cv2, np, numbers)
+def cv_rotate(image, rmatrix, order, scale, missing, image_center, recenter):
```

Comment:
I think we should add this as an option to the built in affine_transform function and not just as an example :)

Reply:
Done.

Reply:
Does that mean this example should be removed?